pkg/operator/sync.go

```diff
@@ -341,39 +347,90 @@ func (optr *Operator) syncMachineConfigController(config *renderConfig) error {
 		}
 	}
 
-	for _, path := range []string{
-		"manifests/machineconfigcontroller/events-rolebinding-default.yaml",
-		"manifests/machineconfigcontroller/events-rolebinding-target.yaml",
-	} {
+	for _, path := range paths.roleBindings {
+		rbBytes, err := renderAsset(config, path)
+		if err != nil {
+			return err
+		}
+		rb := resourceread.ReadRoleBindingV1OrDie(rbBytes)
+		_, _, err = resourceapply.ApplyRoleBinding(optr.kubeClient.RbacV1(), rb)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, path := range paths.clusterRoleBindings {
 		crbBytes, err := renderAsset(config, path)
 		if err != nil {
 			return err
 		}
-		crb := resourceread.ReadRoleBindingV1OrDie(crbBytes)
-		_, _, err = resourceapply.ApplyRoleBinding(optr.kubeClient.RbacV1(), crb)
+		crb := resourceread.ReadClusterRoleBindingV1OrDie(crbBytes)
+		_, _, err = resourceapply.ApplyClusterRoleBinding(optr.kubeClient.RbacV1(), crb)
 		if err != nil {
 			return err
 		}
 	}
 
-	crbBytes, err := renderAsset(config, "manifests/machineconfigcontroller/clusterrolebinding.yaml")
-	if err != nil {
-		return err
+	for _, path := range paths.serviceAccounts {
+		saBytes, err := renderAsset(config, path)
+		if err != nil {
+			return err
+		}
+		sa := resourceread.ReadServiceAccountV1OrDie(saBytes)
+		_, _, err = resourceapply.ApplyServiceAccount(optr.kubeClient.CoreV1(), sa)
+		if err != nil {
+			return err
+		}
 	}
-	crb := resourceread.ReadClusterRoleBindingV1OrDie(crbBytes)
-	_, _, err = resourceapply.ApplyClusterRoleBinding(optr.kubeClient.RbacV1(), crb)
-	if err != nil {
-		return err
+
+	for _, path := range paths.secrets {
+		sBytes, err := renderAsset(config, path)
+		if err != nil {
+			return err
+		}
+		s := resourceread.ReadSecretV1OrDie(sBytes)
+		_, _, err = resourceapply.ApplySecret(optr.kubeClient.CoreV1(), s)
+		if err != nil {
+			return err
+		}
 	}
 
-	saBytes, err := renderAsset(config, "manifests/machineconfigcontroller/sa.yaml")
-	if err != nil {
-		return err
+	for _, path := range paths.daemonsets {
+		dBytes, err := renderAsset(config, path)
+		if err != nil {
+			return err
+		}
+		d := resourceread.ReadDaemonSetV1OrDie(dBytes)
+		_, updated, err := resourceapply.ApplyDaemonSet(optr.kubeClient.AppsV1(), d)
+		if err != nil {
+			return err
+		}
+		if updated {
+			return optr.waitForDaemonsetRollout(d)
+		}
 	}
-	sa := resourceread.ReadServiceAccountV1OrDie(saBytes)
-	_, _, err = resourceapply.ApplyServiceAccount(optr.kubeClient.CoreV1(), sa)
-	if err != nil {
-		return err
+	return nil
+}
+
+func (optr *Operator) syncMachineConfigController(config *renderConfig) error {
+	paths := manifestPaths{
```

Comment:
could these paths be consts? they are never going to change

Reply:
Do you mean having a different const path for each path string?

Reply:
yeah looking it at it feels like they should be consts and could be moved out of the funcs to clean them up a bit. always seemed weird to have the hardcoded paths declared in func in this file.

wdyt @cheesesashimi

Reply:
Completely agree, just thought it would also be a little awkward to make a separate variable for every path but then still have to put them all in the struct

Reply:
I agree with making the paths constants. It allows for reuse and contributes to making tests more robust when the same constant is referenced in a test instead of hard-coding it in two separate places. In terms of organizing them, I suggest grouping them together like this:

```golang
// MachineConfigDaemon Manifest Paths
const (
  mcdClusterRoleManifestPath string = "manifests/machineconfigdaemon/clusterrole.yaml"
  // ...
)

// MachineConfigController Manifest Paths
const (
  mccClusterRoleManifestPath string = "manifests/machineconfigcontroller/clusterrole.yaml"
  // ...
)

// MachineConfigOperator Manifest Paths
const (
  mcoClusterRoleManifestPath string = "manifests/machineconfigoperator/clusterrole.yaml"
  // ...
)
```

Reply:
What about:
```
var machineConfigDaemonManifestPaths = manifestPaths{
	clusterRoles: []string{
		"manifests/machineconfigdaemon/clusterrole.yaml",
		"manifests/machineconfigdaemon/events-clusterrole.yaml",
	},
	roleBindings: []string{
		"manifests/machineconfigdaemon/events-rolebinding-default.yaml",
		"manifests/machineconfigdaemon/events-rolebinding-target.yaml",
	},
	clusterRoleBindings: []string{
		"manifests/machineconfigdaemon/clusterrolebinding.yaml",
	},
	serviceAccounts: []string{
		"manifests/machineconfigdaemon/sa.yaml",
	},
	daemonset: "manifests/machineconfigdaemon/daemonset.yaml",
}
```

And then just use `machineConfigDaemonManifestPaths` in `syncMachineConfigDaemon`?

Reply:
I pushed a commit doing it with var, but I can change it to the const way if making it const is safer. Just think it looks cleaner this way

Reply:
I think they are much better moved, but let's us as consts